### PR TITLE
[Behat] Pages organization fix and ProductContexts cleanup

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

We have decided to divide pages into directories  `Admin/` - for backend and `Shop/` - for frontend.

Moreover I've done a fix of typehinting in the ProductContexts.

### CODE REVIEW

File: src/Sylius/Behat/spec/Page/Shop/Cart/CartSummaryPageSpec.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ function let(Session $session, RouterInterface $router)
 
     function it_is_initializable()
     {
-        $this->shouldHaveType('Sylius\Behat\Page\Cart\CartSummaryPage');
+        $this->shouldHaveType('Sylius\Behat\Page\Cart\Shop\CartSummaryPage');
```
Comment (lchrusciel) [inline src/Sylius/Behat/spec/Page/Shop/Cart/CartSummaryPageSpec.php]: Won't it be better to have just SummaryPage?

File: src/Sylius/Behat/spec/Page/Shop/Checkout/SecurityStepSpec.php [hunk 2]
```diff
@@ -29,12 +29,12 @@ function let(Session $session, RouterInterface $router)
 
     function it_is_initializable()
     {
-        $this->shouldHaveType('Sylius\Behat\Page\Checkout\CheckoutSecurityStep');
+        $this->shouldHaveType('Sylius\Behat\Page\Shop\Checkout\SecurityStep');
```
Comment (lchrusciel) [inline src/Sylius/Behat/spec/Page/Shop/Checkout/SecurityStepSpec.php]: Maybe some better naming? `LoginSep` / `AuthenticationStep`?

File: etc/behat/services/contexts/ui.xml [hunk 3]
```diff
@@ -29,38 +29,38 @@
...
-            <argument type="service" id="sylius.behat.page.checkout_security" />
-            <argument type="service" id="sylius.behat.page.checkout_addressing" />
-            <argument type="service" id="sylius.behat.page.checkout_shipping" />
-            <argument type="service" id="sylius.behat.page.checkout_payment" />
-            <argument type="service" id="sylius.behat.page.checkout_finalize" />
-            <argument type="service" id="sylius.behat.page.checkout_thank_you" />
-            <argument type="service" id="sylius.behat.page.order_payments" />
+            <argument type="service" id="sylius.behat.page.shop.checkout_security" />
+            <argument type="service" id="sylius.behat.page.shop.checkout_addressing" />
+            <argument type="service" id="sylius.behat.page.shop.checkout_shipping" />
```
Comment (lchrusciel) [inline etc/behat/services/contexts/ui.xml]: WDYT about `sylius.behat.page.shop. checkout.shipping`? It is not `CheckoutShippingStep`. we have `ShippingStep` in checkout folder

File: src/Sylius/Behat/spec/Page/Shop/Checkout/SecurityStepSpec.php (hunk 2 above)
Comment (TheMadeleine) [inline src/Sylius/Behat/spec/Page/Shop/Checkout/SecurityStepSpec.php]: since we have the `SecurityService` I would keep this one.

Comment (lchrusciel) [inline src/Sylius/Behat/spec/Page/Shop/Checkout/SecurityStepSpec.php]: We can rename this service as well ;)

**Comment by pjedrzejewski** [comment]
Perfect, thanks Magda! :+1:

**Pull request merged**